### ISSUE TITLE
参考官方示例打印不出loss

### ISSUE THREAD

**Issue by PhenixZhang** [PaddlePaddle/Paddle#38212]
- 版本、环境信息：
    1）PaddlePaddle版本：paddle 2.2.1
    2）GPU：GPU: V100 CUDA 10.1
    4）系统环境：ubuntu 18.04 python 3.7.9

- 训练信息
    1）单机单卡
    2）显存信息
    3）Operator信息

- 问题描述：
参考官方示例：https://www.paddlepaddle.org.cn/tutorials/projectdetail/2182025
运行如下代码发现loss无法正确输出（显示为nan），不清楚具体原因，望解答：
def train(model):
    model.train()
    train_loader = load_data('train')
    opt = paddle.optimizer.SGD(learning_rate=0.01, parameters=model.parameters())
    EPOCH_NUM = 10
    IMG_ROWS, IMG_COLS = 28, 28

    for epoch_id in range(EPOCH_NUM):
        for batch_id, data in enumerate(train_loader()):
            images, labels = data
            images = paddle.to_tensor(images)
            labels = paddle.to_tensor(labels)
            
            predicts = model(images)
            
            loss = F.square_error_cost(predicts, labels)
            avg_loss = paddle.mean(loss)

            if batch_id % 200 == 0:
                print("epoch: {}, batch: {}, loss is: {}".format(epoch_id, batch_id, avg_loss.numpy()))
            
            avg_loss.backward()
            opt.step()
            opt.clear_grad()

    paddle.save(model.state_dict(), 'mnist.pdparams')

%time
model = MNIST()
train(model)

**Comment by joey12300** [comment]
@PhenixZhang 你好，请问你是在aistudio上跑的吗？

**Comment by PhenixZhang** [comment]
不是的 是在本机上跑的～

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by an1018** [comment]
@PhenixZhang 你好，如果使用CNN训练需要添加reshape操作，官方示例中有说明，添加后再训练看是否还有问题
<img width="700" alt="image" src="https://user-images.githubusercontent.com/19808900/146732989-109cc552-1418-4463-8bac-a1d07f5c28b2.png">

**Comment by PhenixZhang** [comment]
好嘞 感谢～
